pkg/virtctl/imageupload/imageupload_test.go

```diff
@@ -700,6 +703,81 @@ var _ = Describe("ImageUpload", func() {
 			validatePVCDefaultInstancetypeLabels()
 		})
 
+		It("Should create DataSource pointing to the PVC", func() {
+			testInit(http.StatusOK)
+			cmd := clientcmd.NewRepeatableVirtctlCommand(
+				commandName, "dv", targetName,
+				"--size", dvSize,
+				"--uploadproxy-url", server.URL,
+				"--insecure",
+				"--force-bind",
+				"--datasource",
+				"--image-path", imagePath,
+			)
+			Expect(cmd()).To(Succeed())
+			Expect(pvcCreateCalled.IsTrue()).To(BeTrue())
+			Expect(kvtesting.FilterActions(&cdiClient.Fake, "create", "datasources")).To(HaveLen(1))
+
+			ds, err := cdiClient.CdiV1beta1().DataSources(targetNamespace).Get(context.Background(), targetName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ds.Spec.Source.PVC).ToNot(BeNil())
+			Expect(ds.Spec.Source.PVC.Name).To(Equal(targetName))
+			Expect(ds.Spec.Source.PVC.Namespace).To(Equal(targetNamespace))
+			Expect(ds.Labels).ToNot(HaveKey(instancetypeapi.DefaultInstancetypeLabel))
+			Expect(ds.Labels).ToNot(HaveKey(instancetypeapi.DefaultInstancetypeKindLabel))
+			Expect(ds.Labels).ToNot(HaveKey(instancetypeapi.DefaultPreferenceLabel))
+			Expect(ds.Labels).ToNot(HaveKey(instancetypeapi.DefaultPreferenceKindLabel))
```

Comment:
`ToNot` is wrong here? It should have the labels?

Reply:
I don't think it's wrong. When we invoke this command here:
```
			testInit(http.StatusOK)
			cmd := clientcmd.NewRepeatableVirtctlCommand(
				commandName, "dv", targetName,
				"--size", dvSize,
				"--uploadproxy-url", server.URL,
				"--insecure",
				"--force-bind",
				"--datasource",
				"--image-path", imagePath,
			)
```
This is not passing any key/values of the labels, because we expect to test here only the creation of the DataSource that is pointing to the PVC.

Reply:
Let's add these labels, because we should expect from the user to create DataSource with them.

Reply:
We should add tests for all features we introduce, so setting labels should be tested here as well. Thanks for adding it.

Reply:
Yeah it's good to test it, because labels of this object was nil, which could lead to a bug.